### ISSUE TITLE
Non trivial question regarding mini_batch size 

### ISSUE THREAD

**Issue by arnaud-nt2i** [AlexeyAB/darknet#6623]
@AlexeyAB , @WongKinYiu , @kmsravindra, @cenit, @stephanecharette
Hi everyone!

According to the wiki section:  
> GPU processes mini_batch samples at once, and the weights will be updated for batch samples

So the batch size is the number of samples processed before the weighs update. 
The literature is full of discussion regarding picking the wright batch size (sometimes called mini-batch) for training.

In Yolo, the mini_batch refers to something else: 
> mini_batch = batch/subdivisions

and seems to only be a computational trick to avoid loading and processing the batch number of images in one time and should not have an impact on mAP.

So why did AlexeyAB said on issue #4695 and #4386:
> mini_batch size is more important than batch size

> Higher mini_batch -> higher accuracy mAP/Top1/Top5

For me, it makes no sense and I really would like to understand this,
an explanation could be via referring to code, text, or paper.

Thanks a lot!

**Comment by AlexeyAB** [comment]
1. mini_batch is a number of samples for one `forward->backward`. batch_size is a number of samples for `weights_upate`. 
    batch_size >= mini_batch

2. mini_batch and batch_size are different values in all frameworks: Darknet, Pytorch, TensorFlow... 
    For example, https://github.com/pytorch/examples/blob/master/imagenet/main.py#L148-L150

    * if you will train in Pytorch with `torch.nn.parallel.DistributedDataParallel` with `torch.utils.data.DataLoader(batch_size = 16, ...` and **without** `torch.nn.SyncBatchNorm` on several 8 x GPUs, then actually will be `batch_size=128 mini_batch=16`

    * if you will train in Pytorch with `torch.nn.parallel.DistributedDataParallel` with `torch.utils.data.DataLoader(batch_size = 16, ...` and **with** `torch.nn.SyncBatchNorm` on several 8 x GPUs, then actually will be `batch_size=128 mini_batch=128`

3. Higher mini_batch -> higher accuracy mAP/Top1/Top5

**Comment by arnaud-nt2i** [comment]
@AlexeyAB ok thanks a lot for your answer things are a bit clearer now!
To everyone reading this, I also recommend the BN and CBN papers for a good understanding:
[https://arxiv.org/abs/1502.03167](url)
[https://arxiv.org/abs/2002.05712](url)

and this topic:  https://github.com/AlexeyAB/darknet/issues/4386#issuecomment-589754020
